snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -292,6 +294,29 @@ def _ensure_base(self) -> None:
             )
             self.clean_project()
 
+    def _setup_environment(self, *, tempfile_func=tempfile.NamedTemporaryFile) -> None:
```

Comment:
I assume tempfile_func is for testing purposes.  Any reason it can't just be mocked  (and verify `mock_calls`)?

Reply:
This is much nicer I believe.
Also, mocking NamedTemporaryDirectory is not a nice task. I did it, and preferred this which is much more straightforward.